### PR TITLE
op_selection 2/ add test for implicit alias and disconnected selection

### PR DESCRIPTION
**Pull request by yuhan**
## Checklist

- [ ] My change requires a change to the documentation and I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: python_modules/dagster/dagster_tests/core_tests/graph_tests/test_op_selection.py [hunk 1]
```diff
@@ -261,38 +250,37 @@ def _aliased():
...
+    @job
+    def _reuse_ops_job():
+        add_one(return_one())
+        add_one(return_one())
+        add_one(return_one())
+
+    result_1 = _reuse_ops_job.execute_in_process(op_selection=["return_one*"])
+
+    assert result_1.success
+    assert len(_success_step_keys(result_1)) == 2
```
Comment (dpeng817) [inline python_modules/dagster/dagster_tests/core_tests/graph_tests/test_op_selection.py:275]: shouldn't this be 3? Or does the * imply that there's always more (doesn't include the empty char case)?

File: python_modules/dagster/dagster_tests/core_tests/graph_tests/test_op_selection.py [hunk 2]
```diff
@@ -261,38 +250,37 @@ def _aliased():
...
+    assert result_2.success
     assert (
         len(
             [
                 evt.step_key
-                for evt in result_for_subset.all_node_events
+                for evt in result_2.all_node_events
                 if evt.event_type == DagsterEventType.STEP_SUCCESS
             ]
         )
```
Comment (dpeng817) [inline python_modules/dagster/dagster_tests/core_tests/graph_tests/test_op_selection.py:286]: shouldn't this assertion be checking exact length value? I think you can also reuse _success_step_keys here, no?

**Review by dpeng817 — DISMISSED** [review]
Some relatively minor questions about some of the tests, feel free to dismiss if I'm off mark here.

File: python_modules/dagster/dagster_tests/core_tests/graph_tests/test_op_selection.py (hunk 1 above)
Comment (yuhan) [reply python_modules/dagster/dagster_tests/core_tests/graph_tests/test_op_selection.py:275]: it should be executing `return_one` and `add_one`. the remaining ops are: `["add_one_1", "return_one_1", "add_one_2", "return_one_2"]`

**Review by dpeng817 — APPROVED** [review]

**Pull request merged**